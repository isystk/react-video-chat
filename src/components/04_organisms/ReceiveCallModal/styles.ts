import { css, keyframes } from '@emotion/css'

const useStyles = (theme) => css`{
  notion: {
    width: '300px',
    '& .myHeadPhoto': {
      width: '100px',
      height: '100px',
      borderRadius: '50%',
      margin: '0 auto',
      overflow: 'hidden',
    },
    '& .myHeadPhoto img': {
      width: '100%',
    },
    '& .myName': {
      width: '100%',
      fontSize: '24px',
      color: '#222222',
      textAlign: 'center',
      marginTop: '15px',
    },
    '& .btn': {
      width: '100%',
      textAlign: 'center',
    },
    '& .btn button': {
      margin: '25px 25px 5px',
    },
    '& .loading': {
      margin: '20px auto 0',
      width: '16px',
    },
  },
}`

export { useStyles }
