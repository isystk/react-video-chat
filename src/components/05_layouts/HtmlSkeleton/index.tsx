import React, { Children, FC, useState } from 'react'
import { connect } from '@/components/hoc'
import { ContainerProps, WithChildren } from 'types'
import useAppRoot from '@/stores/useAppRoot'
import Header from '@/components/04_organisms/Header'
import SideMenu from '@/components/04_organisms/SideMenu'
import Title, { TitleProps } from './Title'
import Head from 'next/head'
import { isReactElement } from '@/utils/general/object'
import { CssBaseline } from '@material-ui/core'

/** HtmlSkeleton Props */
export type HtmlSkeletonProps = WithChildren
/** Presenter Props */
export type PresenterProps = HtmlSkeletonProps & {
  title?: TitleProps['children']
  description: string
}

/** Presenter Component */
const HtmlSkeletonPresenter: FC<PresenterProps> = ({
  main,
  title,
  description,
  isMenuOpen,
  setMenuOpen,
  children,
  ...props
}) => (
  <>
    <Head>
      {/* タイトル */}
      <title>{title}</title>
      {/* favicon */}
      <link rel="icon" href="/favicon.ico" />
      {/* ブラウザテーマカラー */}
      <meta name="theme-color" content="#ffffff" />
      {/* サイト概要 */}
      <meta name="description" content={description} />
      {/* OGP 画像URL */}
      <meta property="og:image" content={'/ogp-image.png'} />
      {/* OGP タイトル */}
      <meta name="og:title" content={title} />
      {/* OGP サイト概要 */}
      <meta name="og:description" content={description} />
      {/* OGP Twitterカード */}
      <meta name="twitter:card" content="summary" />
      {/* apple ポータブル端末 アイコン */}
      <link rel="apple-touch-icon" href="/apple-touch-icon.png" />
      {/* manifest.json */}
      <link rel="manifest" href="/manifest.json" />
    </Head>
    <Context.Provider value={main}>
      <CssBaseline />
      <Header isMenuOpen={isMenuOpen} setMenuOpen={setMenuOpen} />
      <div>{children}</div>
      <SideMenu isMenuOpen={isMenuOpen} setMenuOpen={setMenuOpen} />
    </Context.Provider>
  </>
)

/** Container Component */
const HtmlSkeletonContainer: React.FC<
  ContainerProps<HtmlSkeletonProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  let title: TitleProps['children'] | undefined = undefined
  const main = useAppRoot()
  const [isMenuOpen, setMenuOpen] = useState(false)

  if (!main) return <></>

  const APP_NAME = process.env.APP_NAME

  children = Children.map(children, (child) =>
    isReactElement(child) && child.type === Title
      ? (title = `${child.props.children} | ${APP_NAME}`) && undefined
      : React.cloneElement(child)
  )
  if (!title) title = process.env.APP_NAME

  const description = `${APP_NAME} is an application that allows you to casually chat with your friends!`
  return presenter({
    main,
    title,
    description,
    isMenuOpen,
    setMenuOpen,
    children,
    ...props,
  })
}

export const Context = React.createContext(null)

export default connect<HtmlSkeletonProps, PresenterProps>(
  'HtmlSkeleton',
  HtmlSkeletonPresenter,
  HtmlSkeletonContainer
)

// Sub Component
export type { TitleProps }
export { Title }
