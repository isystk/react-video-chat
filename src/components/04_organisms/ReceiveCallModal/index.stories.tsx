import { Meta, Story } from '@storybook/react'
import React from 'react'
import ReceiveCallModal from './index'
import MainService from '@/services/main'
import { Context } from '@/components/05_layouts/HtmlSkeleton'

export default {
  title: '04_organisms/ReceiveCallModal',
  component: ReceiveCallModal,
} as Meta

const Template: Story = (props) => {
  const main = new MainService(() => ({}))
  main.setName('isystk')
  main.setRoomId('test')
  main.addMember({
    connectionId: 'bbb',
    name: 'bbb',
    photo: 'images/friends/David.png', 
  })
  main.video.members = [
    {
      connectionId: 'bbb',
      name: 'bbb',
      photo: 'images/friends/David.png',
    },
  ]
  main.video.nowCallReceiving= true
  return (
    <Context.Provider value={main}>
      <ReceiveCallModal/>
    </Context.Provider>
  )
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
