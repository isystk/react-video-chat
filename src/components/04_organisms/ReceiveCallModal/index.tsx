import React, {FC, useContext} from 'react'
import Modal from '@/components/01_atoms/Modal'
import { Button, CssBaseline } from '@material-ui/core'
import Container from '@material-ui/core/Container'
import { ContainerProps, WithChildren } from 'types'
import { useStyles } from './styles'
import { connect } from '@/components/hoc'
import {Context} from "@/components/05_layouts/HtmlSkeleton";
import MainService from "@/services/main";
import * as _ from "lodash";

/** ReceiveCallModalProps Props */
export type ReceiveCallModalProps = WithChildren
/** Presenter Props */
export type PresenterProps = ReceiveCallModalProps & {
  main
  classes
  isOpen
  connectionId
  name
  photo
}

/** Presenter Component */
const ReceiveCallModalPresenter: FC<PresenterProps> = ({
  main,
  classes,
  isOpen,
  connectionId,
  name,
  photo,
  ...props
}) => (
  <>
    <Modal isOpen={isOpen} hideCloseBtn={true}>
      <Container component="main">
        <CssBaseline />
        <div className="notion">
          <div className="myHeadPhoto">
            <img src={photo} alt="" />
          </div>
          <div className="myName">{name}</div>
          <div className="btn">
            <Button
              color="secondary"
              onClick={(e) => main.video.sendRejectCall(connectionId)}
              type="submit"
              variant="contained"
            >
              また後で
            </Button>
            <Button
              color="primary"
              onClick={(e) => main.video.sendAcceptCall(connectionId)}
              type="submit"
              variant="contained"
            >
              いいよ！
            </Button>
          </div>
          <div className="loading">
            <div className="snippet" data-title=".dot-pulse">
              <div className="stage">
                <div className="dot-pulse"></div>
              </div>
            </div>
          </div>
        </div>
      </Container>
    </Modal>
  </>
)

/** Container Component */
const ReceiveCallModalContainer: React.FC<
  ContainerProps<ReceiveCallModalProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  const main = useContext<MainService | null>(Context)
  if (!main) return <></>
  const classes = useStyles()

  if (_.size(main.video.members) === 0) return <></>

  const isOpen = main.video.nowCallReceiving

  const { connectionId, name, photo } = main.video.members[0]
  return presenter({
    children,
    main,
    classes,
    isOpen,
    connectionId,
    name,
    photo,
    ...props,
  })
}

export default connect<ReceiveCallModalProps, PresenterProps>(
  'ReceiveCallModal',
  ReceiveCallModalPresenter,
  ReceiveCallModalContainer
)
