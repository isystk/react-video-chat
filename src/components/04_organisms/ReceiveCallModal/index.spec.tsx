import React from 'react'
import renderer from 'react-test-renderer'
import ReceiveCallModal from './index'
import '@testing-library/jest-dom/extend-expect'
import { Context } from '@/components/05_layouts/HtmlSkeleton'
import MainService from '@/services/main'

describe('ReceiveCallModal', () => {
  it('Match Snapshot', () => {
    const main = new MainService(() => ({}))
    main.setName('isystk')
    main.room.rooms = [{ id: 1, name: 'test', description: 'test' }]
    main.room.setRoomId(1)
    main.addMember({
      connectionId: 'bbb',
      name: 'bbb',
      photo: 'images/friends/David.png',
    })
    main.video.members = [
      {
        connectionId: 'bbb',
        name: 'bbb',
        photo: 'images/friends/David.png',
      },
    ]
    main.video.nowCallReceiving = true
    const component = renderer.create(
      <Context.Provider value={main}>
        <ReceiveCallModal />
      </Context.Provider>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
