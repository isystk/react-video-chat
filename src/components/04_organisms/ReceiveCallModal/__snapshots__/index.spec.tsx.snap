// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReceiveCallModal Match Snapshot 1`] = `null`;
